src/votekit/election_types.py

```diff
@@ -1,117 +1,140 @@
+            for candidate in fp_order:
+                if fp_votes[candidate] >= self.threshold:
+                    elected.append(candidate)
+                    remaining.remove(candidate)
+                    ballots = self.transfer(
+                        candidate, ballots, fp_votes, self.threshold
+                    )
+        # since no one has crossed threshold, eliminate one of the people
+        # with least first place votes
+        elif self.next_round():
             lp_votes = min(fp_votes.values())
             lp_candidates = [
-                candidate for candidate, votes in fp_votes.items() if votes == lp_votes
+                candidate for candidate in fp_order if fp_votes[candidate] == lp_votes
             ]
             # is this how to break ties, can be different based on locality
             lp_cand = random.choice(lp_candidates)
+            eliminated.append(lp_cand)
             ballots = remove_cand(lp_cand, ballots)
-            candidates.remove(lp_cand)
-            self.eliminated.add(lp_cand)
-
-        return PreferenceProfile(ballots=ballots), Outcome(
-            elected=self.elected,
-            eliminated=self.eliminated,
-            remaining=set(candidates),
-            votes=fp_votes,
+            remaining.remove(lp_cand)
+
+        self.election_state = ElectionState(
+            curr_round=self.election_state.curr_round + 1,
+            elected=elected,
+            eliminated=eliminated,
+            remaining=remaining,
+            profile=PreferenceProfile(ballots=ballots),
+            previous=self.election_state,
         )
+        return self.election_state
 
-    def run_election(self) -> Outcome:
+    def run_election(self) -> ElectionState:
         """
         Runs complete STV election
         """
-        profile = deepcopy(self.profile)
-
         if not self.next_round():
             raise ValueError(
                 f"Length of elected set equal to number of seats ({self.seats})"
             )
 
         while self.next_round():
-            profile, outcome = self.run_step(profile)
+            self.run_step()
+
+        return self.election_state
 
-        return outcome
+    def get_init_profile(self):
+        "returns the initial profile of the election"
+        return self.__profile
 
 
 ## Election Helper Functions
 
 
-def compute_votes(candidates: list, ballots: list[Ballot]) -> dict:
+def compute_votes(candidates: list, ballots: list[Ballot]) -> tuple[dict, list]:
     """
     Computes first place votes for all candidates in a preference profile
     """
     votes = {}
-
     for candidate in candidates:
         weight = Fraction(0)
         for ballot in ballots:
             if ballot.ranking and ballot.ranking[0] == {candidate}:
                 weight += ballot.weight
         votes[candidate] = weight
 
-    return votes
+    order = [y[0] for y in sorted(votes.items(), key=lambda x: x[1], reverse=True)]
```

Comment:
I wonder if this is the best way to do this, effectively returning the elements twice.  Might consider creating a new dictionary with correct order, or return a list with (k, v) tuples, instead of both.

Reply:
Ok I've changed this by just returning the list and eliminating the use of the fp_order list everywhere